src/methods/voronoi.jl

```diff
@@ -0,0 +1,210 @@
+
+GeometryOps.jl provides a method for computing the Voronoi tessellation of a set of points,
+using the [DelaunayTriangulation.jl](https://github.com/JuliaGeometry/DelaunayTriangulation.jl) package.
+
+Right now, the GeometryOps.jl method can only provide clipped voronoi tesselations, as the function returns a list of GeoInterface polygons.
+If you need an unbounded tessellation, open an issue and we can discuss the best way to represent unbounded polygons within GeometryOps.
+
+## Example
+
+### Simple tessellation
+```@example simple
+import GeometryOps as GO, GeoInterface as GI
+using CairoMakie # to plot
+
+points = tuple.(randn(20), randn(20))
+polygons = GO.voronoi(points)
+f, a, p = plot(polygons[1]; label = "Voronoi cell 1")
+for (i, poly) in enumerate(polygons[2:end])
+    plot!(a, poly; label = "Voronoi cell $(i+1)")
+end
+scatter!(a, points; color = :black, markersize = 10, label = "Generators")
+axislegend(a)
+f
+```
+
+## Implementation
+
+This implementation mainly just preforms some assertion checks before passing the Arguments
+to the DelaunayTriangulation package. We always set the argument `clip` to the DelaunayTriangulation
+`voronoi` call to `True` such that we can return a list of valid polygons. The default clipping polygon
+is the convex hull of the tessleation, but the user can pass in a bounding polygon with the `clip_polygon`
+argument. After the call to `voronoi`, the call then unpacks the voronoi output into GeoInterface
+polygons, whose point match the float type input by the user.
+=#
+
+struct __NoCRSProvided end
+
+"""
+    voronoi(geometries, [T = Float64]; clip_polygon = nothing, kwargs...)
+
+Compute the Voronoi tessellation of the points in `geometries`.
+Returns a vector of `GI.Polygon` objects representing the Voronoi cells,
+in the same order as the input points.
+
+## Arguments
+- `geometries`: Any GeoInterface-compatible geometry or collection of geometries
+  that can be decomposed into points
+- `T`: Float-type for returned polygons points (default: Float64)
+
+## Keyword Arguments  
+- `clip_polygon`: what bounding shape should the Voronoi cells be clipped to? (default: nothing -> clipped to the convex hull)
+    clip_polygon can of several types: (1) a GeoInterface polygon, (2) a two-element tuple where the first element is a list of tuple points
+    and the second element is a list of integer indices to indicate the order of the provided points, or (3) a a two-element tuple where the
+    first element is a tuple of tuple points and the second element is a tuple of integer indices to indicate the order of the provided points
+- $CRS_KEYWORD
+- `rng`: random number generator to generating the voronoi tesselation
+
+!!! warning
+    This interface only computes the 2-dimensional Voronoi tessellation!
+    Only clipped voronoi tesselations can be created!
+    Only `T = Float64` is guaranteed good results by underlying package DelaunayTriangulation.
```

Comment:
> Only `T = Float64` is guaranteed good results

I think since DelaunayTriangulation v1.1 this is no longer true FYI, `Float32` is also supported as long as the `AdaptiveKernel()` is used (which is the default). Maybe I've forgotten to loosen warnings about `Float32` in the documentation or have you run into some counter-examples?

Reply:
I saw that in the documentation and thus included it here. That is good to know though! I will remove this warning!

Reply:
```suggestion
    Only `T = Float64` or `Float32` are guaranteed good results by the underlying package DelaunayTriangulation.
```